src/OpenTelemetry.Api/Trace/Status.cs

```diff
@@ -142,7 +142,12 @@ namespace OpenTelemetry.Trace
         /// </summary>
         public static readonly Status DataLoss = new Status(StatusCanonicalCode.DataLoss);
 
-        internal Status(StatusCanonicalCode statusCanonicalCode, string description = null)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Status"/> struct.
+        /// </summary>
+        /// <param name="statusCanonicalCode">Canonical result code of span execution.</param>
+        /// <param name="description">Status description.</param>
+        public Status(StatusCanonicalCode statusCanonicalCode, string description = null)
```

Comment:
I think the goal with the `internal` was to encourage re-use of the `static` instances to prevent unnecessary allocations. Would you be open to refactoring to lookup from a dictionary? Something like...

```csharp
public static readonly IDictionary<StatusCanonicalCode, Status> Definitions { get; } = new Dictionary<StatusCanonicalCode, Status>();

internal Status(StatusCanonicalCode statusCanonicalCode, string description = null)
{
            this.CanonicalCode = statusCanonicalCode;
            this.Description = description;
            this.IsValid = true;

            Definitions[statusCanonicalCode] = this;
}
```

The idea being as the `static` instances are created they will be added to the `Definitions` which you can then use for lookup. Make sure `Definitions` is defined before the others so it is initialized first.

Reply:
@CodeBlanch 

I changed the signature of this `constructor` to set the description. Proposed changes will not help set the description. By keeping this constructor as `internal` we could set only the `statusCanonicalCode` by calling `Status.Cancelled` or `Status.Ok`, etc.,).

```
            var activity = source.StartActivity("Test Activity");
            activity.SetStatus(Status.Cancelled);
            activity?.Stop();
```
Should we consider splitting the method arguments for `SetStatus` into two. Like `StatusCanonicalCode` and `description`?

Reply:
@rajkumar-rangaraj Gotcha! Yes I think maybe we should split the arguments. Something like this:
```csharp
        public static void SetStatus(this Activity activity, Status status, string description = null)
        {
            activity?.AddTag(SpanAttributeConstants.StatusCodeKey, SpanHelper.GetCachedCanonicalCodeString(status.CanonicalCode));
            activity?.AddTag(SpanAttributeConstants.StatusDescriptionKey, description ?? status.Description);
        }
```
Basically so we can do the set in an allocation-free way. Kind of brings up an interesting question, would we ever want to set status code but not status description? If not, I think we're good.

For the `GetStatus` method can you switch to the `WithDescription` method? That will reuse the defined instance if the description is the standard one.

Should be able to leave the ctor `internal` with those changes.